modules/apps/object/object-dynamic-data-mapping-form-field-type/src/main/java/com/liferay/object/dynamic/data/mapping/form/field/type/internal/attachment/AttachmentDDMFormFieldTemplateContextContributor.java

```diff
@@ -57,39 +68,96 @@
 		).put(
 			"objectEntryId", ddmFormField.getProperty("objectEntryId")
 		).put(
-			"uploadURL",
-			_getUploadURL(
-				ddmFormField,
-				ddmFormFieldRenderingContext.getHttpServletRequest())
+			"url", _getURL(ddmFormField, ddmFormFieldRenderingContext)
 		).put(
 			"value", ddmFormFieldRenderingContext.getValue()
 		).build();
 	}
 
-	private String _getUploadURL(
-		DDMFormField ddmFormField, HttpServletRequest httpServletRequest) {
+	private long _getGroupId(
+		DDMFormFieldRenderingContext ddmFormFieldRenderingContext) {
 
-		String uploadURL = GetterUtil.getString(
-			ddmFormField.getProperty("uploadURL"));
+		long groupId = GetterUtil.getLong(
+			ddmFormFieldRenderingContext.getProperty("groupId"));
 
-		if (Validator.isNotNull(uploadURL)) {
-			return uploadURL;
+		if (groupId != 0) {
+			return groupId;
 		}
 
+		HttpServletRequest httpServletRequest =
+			ddmFormFieldRenderingContext.getHttpServletRequest();
+
+		ThemeDisplay themeDisplay =
+			(ThemeDisplay)httpServletRequest.getAttribute(
+				WebKeys.THEME_DISPLAY);
+
+		return themeDisplay.getCompanyGroupId();
+	}
+
+	private String _getItemSelectorURL(
+		long groupId, String portletNamespace,
+		RequestBackedPortletURLFactory requestBackedPortletURLFactory) {
+
+		FileItemSelectorCriterion fileItemSelectorCriterion =
+			new FileItemSelectorCriterion();
+
+		fileItemSelectorCriterion.setDesiredItemSelectorReturnTypes(
+			new FileEntryItemSelectorReturnType());
+
+		PortletURL itemSelectorURL = _itemSelector.getItemSelectorURL(
+			requestBackedPortletURLFactory,
+			_groupLocalService.fetchGroup(groupId), groupId,
+			portletNamespace + "selectAttachmentEntry",
+			fileItemSelectorCriterion);
+
+		return itemSelectorURL.toString();
+	}
+
+	private String _getURL(
+		DDMFormField ddmFormField,
+		DDMFormFieldRenderingContext ddmFormFieldRenderingContext) {
+
+		String url = GetterUtil.getString(ddmFormField.getProperty("url"));
+
+		if (Validator.isNotNull(url)) {
```

Comment:
This will be true after an evaluation, right? Do you remember a conversation we've had about this that I asked you what happens if a user changes the state of the `url` property in the Attachment React component and the evaluation is called again?

Reply:
Yeah, but from what I remember we decided to keep this way because the same could happen for other properties.. like the `acceptedFileExtensions`...

Also if I remove this code I wouldn't be able to regenerate the URL after evaluation because I need the `portletId`, `folderId` and `objectFieldId`. I could if I sent these properties through the `getParameters` method (this way they will be saved in the form context) but the same problem can happen as the user will be able to change the value of folderId, for example.

Reply:
Got it. It makes sense. Let's leave it like that. Thanks for reminding me, @carolmariaabb